core/ledger/ledger.go

```diff
@@ -159,7 +159,25 @@ func (ledger *Ledger) CommitTxBatch(id interface{}, transactions []*protos.Trans
 	writeBatch := gorocksdb.NewWriteBatch()
 	defer writeBatch.Destroy()
 	block := protos.NewBlock(transactions, metadata)
-	block.NonHashData = &protos.NonHashData{}
+
+	ccEvents := []*protos.ChaincodeEvent{}
+
+	if transactionResults != nil {
+		ccEvents = make([]*protos.ChaincodeEvent, len(transactionResults))
+		for i := 0; i < len(transactionResults); i++ {
```

Comment:
I thought over that a bit....I like that I know exactly what i'm getting when iterating old fashioned way (definitely no extra instructions). For this purpose its exactly what's needed - nothing more or less.